components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/PublishedDataSetEventModelEx.cs

```diff
@@ -34,15 +34,13 @@ public static class PublishedDataSetEventsModelEx {
                         .ToList(),
                     WhereClause = publishedEvent.WhereClause.Clone(),
                 },
-                AggregateFilter = null,
                 DiscardNew = publishedEvent.DiscardNew,
                 QueueSize = publishedEvent.QueueSize,
                 TriggerId = publishedEvent.TriggerId,
                 MonitoringMode = publishedEvent.MonitoringMode,
                 StartNodeId = publishedEvent.EventNotifier,
                 RelativePath = publishedEvent.BrowsePath,
                 AttributeId = null,
-                DataChangeFilter = null,
                 IndexRange = null,
                 SamplingInterval = null
```

Comment:
Looks like if this is set to null, the GetValueOrDefault methods used on this property will default to 1 second. Whereas for events a client should always set the sampling interval to 0. To enforce this, maybe this is an appropriate place?
[](https://reference.opcfoundation.org/v104/Core/docs/Part4/5.12.1/)

Similar consideration for any other properties? E.g., queue size of 0?

Reply:
Fixed for SamplingInterval. Regarding QueueSize, if that should always be 0 - we shouldn't have it as a setting. Let's find out.